samples/net/eth_native_posix/README.rst

```diff
@@ -0,0 +1,118 @@
+.. _eth-native-posix-sample:
+
+Native Posix Ethernet
+#####################
+
+Overview
+********
+
+The eth_native_posix sample application for Zephyr creates a **zeth** network
+interface to the host system. One can communicate with Zephyr via this network
+interface.
+
+The source code for this sample application can be found at:
+:file:`samples/net/eth_native_posix`.
+
+Building And Running
+********************
+
+To build the eth_native_posix sample application, follow the steps
+below.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/eth_native_posix
+   :board: native_posix
+   :conf: <config file to use>
+   :goals: build
+   :compact:
+
+Normally one needs extra priviliges to create and configure the TAP device in
+the host system. Use sudo to execute the Zephyr process with admin privileges,
+like this:
+
+.. code-block:: console
+
+    sudo --preserve-env=ZEPHYR_BASE make run
+
+If the sudo command reports an error, then try to execute it like this:
+
+.. code-block:: console
+
+    sudo --preserve-env make run
+
+If you want to connect two Zephyr instances together, you can do it like this:
+
+Create two Zephyr config files prj1.conf and prj2.conf. You can use
+:file:`samples/net/eth_native_posix/prj.conf` as a base.
+
+Set prj1.conf IP address configuration like this:
+
+.. code-block:: console
+
+    CONFIG_NET_APP_MY_IPV6_ADDR="2001:db8:100::1"
+    CONFIG_NET_APP_PEER_IPV6_ADDR="2001:db8:100::2"
+    CONFIG_NET_APP_MY_IPV4_ADDR="198.51.100.1"
+    CONFIG_NET_APP_PEER_IPV4_ADDR="198.51.100.2"
+    CONFIG_NET_APP_MY_IPV4_GW="203.0.113.1"
+    CONFIG_ETH_NATIVE_POSIX_RANDOM_MAC=n
+    CONFIG_ETH_NATIVE_POSIX_MAC_ADDR="00:00:5e:00:53:64"
+    CONFIG_ETH_NATIVE_POSIX_SETUP_SCRIPT="echo"
+    CONFIG_ETH_NATIVE_POSIX_DRV_NAME="zeth.1"
+
+Set prj2.conf IP address configuration like this:
+
+.. code-block:: console
+
+    CONFIG_NET_APP_MY_IPV6_ADDR="2001:db8:200::1"
+    CONFIG_NET_APP_PEER_IPV6_ADDR="2001:db8:200::2"
+    CONFIG_NET_APP_MY_IPV4_ADDR="203.0.113.1"
+    CONFIG_NET_APP_PEER_IPV4_ADDR="203.0.113.2"
+    CONFIG_NET_APP_MY_IPV4_GW="198.51.100.1"
+    CONFIG_ETH_NATIVE_POSIX_RANDOM_MAC=n
+    CONFIG_ETH_NATIVE_POSIX_MAC_ADDR="00:00:5e:00:53:c8"
+    CONFIG_ETH_NATIVE_POSIX_SETUP_SCRIPT="echo"
+    CONFIG_ETH_NATIVE_POSIX_DRV_NAME="zeth.2"
+
+Then compile and run two Zephyr instances:
```

Comment:
If you leave the example as is, then add a note here:

    Then compile and run two Zephyr instances (if ``sudo --preserve-env=ZEPHYR_BASE`` 
    gives an error, just use ``sudo --preserve-env`` instead):